src/services/DiscordRpc.cxx

```diff
@@ -0,0 +1,61 @@
+#include "DiscordRpc.hxx"
+#include "LOG.hxx"
+
+#ifdef USE_DISCORD_RPC
+#include "discord.h"
+
+static std::unique_ptr<discord::Core> core;
+#endif // USE_DISCORD_RPC
+
+void DiscordRpc::init()
+{
+#ifdef USE_DISCORD_RPC
+  discord::Core *newcore{};
+  auto result = discord::Core::Create(980219396125523978, DiscordCreateFlags_Default, &newcore);
+  core.reset(newcore);
+  if (!core)
+  {
+    LOG(LOG_ERROR) << "Failed to instantiate discord core! (err " << static_cast<int>(result) << ")";
+    return;
+  }
+
+  core->SetLogHook(discord::LogLevel::Debug, [](discord::LogLevel level, const char *message)
+                         { LOG(LOG_INFO) << "Discord: " << static_cast<uint32_t>(level) << " " << message << "\n"; });
+#endif // USE_DISCORD_RPC
+}
+
+void DiscordRpc::updatePresence(const char *map)
+{
+#ifdef USE_DISCORD_RPC
+  if (!core)
+    return;
+
+  std::string statestr = "Playing map: ";
+  statestr.append(map);
+
+  discord::Activity activity{};
+  activity.SetDetails("Where am I?");
```

Comment:
Consider instead of hard code strings, create some static const string on top of file. So if anyone later needs to change them, can easier navigate.

Reply:
I dont know what write here, it can be add later, when you have more extensive activity in game, like playing map, population, miney in city etc